brave/brave-core
# brave search promotion from NTP

fix https://github.com/brave/brave-browser/issues/23724

<img width="900" alt="Screen Shot 2022-07-08 at 2 35 44 PM" src="https://user-images.githubusercontent.com/6786187/177924098-27cdaea8-908c-4697-9729-e65b944fbae5.png">

Other widgets are hidden when window size is not sufficient to show them with promotion banner.
<img width="600" alt="Screen Shot 2022-07-08 at 2 35 58 PM" src="https://user-images.githubusercontent.com/6786187/177924114-ba5a77d4-1de0-4c39-82f7-8344877f2199.png">


## Submitter Checklist:


## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on

## Test Plan:

1. Launch Brave with `--enable-features=BraveSearchNTP`
2. Open NTP and check promotion banner & searchh box is visible
3. Promotion is hidden when `NTP SI` is background image
4. Click outside of promotion makes it invisible and visible again after reloading
6. When window size gets narrow and promotion popup and other widgets are overlapped much, other widget will be hidden
7. Click close(`x`) button in the popup and check popup is not visible anymore after reloading or creating another NTP
